### PR TITLE
feat: Implementation for Plugin Interface CDK Read Method

### PR DESCRIPTION
**Pull request by mildaniel**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by wchengru — APPROVED** [review]

File: samcli/lib/iac/cfn/helpers.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""
+CFN IaC helper functions
+"""
+from samcli.lib.iac.plugins_interfaces import Stack
+from samcli.lib.providers.sam_base_provider import SamBaseProvider
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+
+
+def get_resolved_cfn_template(template, parameter_overrides, global_parameter_overrides) -> Stack:
```
Comment (torresxb1) [inline samcli/lib/iac/cfn/helpers.py:9]: nit but could we include the types for the parameters?

Comment (mildaniel) [reply samcli/lib/iac/cfn/helpers.py:9]: Yup, good call out.

File: samcli/lib/iac/cdk/cdk_iac.py [hunk 2]
```diff
@@ -31,3 +115,229 @@ def update_packaged_locations(self, stack: Stack) -> bool:
...
+        assets = _collect_assets(ca_stack)
+        LOG.debug("Found assets: %s", str(assets))
+        asset_parameters = {
+            asset_param
+            for asset in assets.values()
+            for asset_param in asset.extra_details.get("assetParameters", {}).values()
+        }
+        sections = {}
+        options_map = self._context.command_options_map
+        template = get_resolved_cfn_template(
```
Comment (hawflau) [inline samcli/lib/iac/cdk/cdk_iac.py:185]: I think it's neat to do the intrinsics resolving here. But how do we get the original template back? That's  something we need to revisit when we start to work on the `write_project` implementation.

Comment (mildaniel) [reply samcli/lib/iac/cdk/cdk_iac.py:185]: I believe that topic is part of the planned discussion we have for `write_project`. There are a few options here, but I don't think this should be something difficult.

File: samcli/lib/iac/cfn/helpers.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+CFN IaC helper functions
+"""
+from typing import Dict
+
+from samcli.lib.iac.plugins_interfaces import Stack
+from samcli.lib.providers.sam_base_provider import SamBaseProvider
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+
+
+def get_resolved_cfn_template(
```
Comment (hawflau) [inline samcli/lib/iac/cfn/helpers.py:11]: Why do we need to define an extra function to just wrap `SamLocalStackProvider.merge_parameter_overrides(...)` and do nothing else?

Comment (hawflau) [inline samcli/lib/iac/cfn/helpers.py:11]: Why do we need to define an extra function to just wrap `SamLocalStackProvider.merge_parameter_overrides(...)` and do nothing else?

**Review by hawflau — CHANGES_REQUESTED** [review]

File: samcli/lib/iac/cdk/cdk_iac.py (hunk 2 above)
Comment (hawflau) [reply samcli/lib/iac/cdk/cdk_iac.py:185]: sounds good. Just wanted to keep a note here as a reminder

File: samcli/lib/iac/cfn/helpers.py (hunk 3 above)
Comment (mildaniel) [reply samcli/lib/iac/cfn/helpers.py:11]: Not really needed. I can remove this.

**Review by hawflau — APPROVED** [review]

**Pull request merged**